src/python/WMCore/Services/Requests.py

```diff
@@ -367,7 +369,9 @@ def cacheTopPath(self, given_path, service_name):
                 break
         else:
             idir = tempfile.mkdtemp(prefix='.wmcore_cache_')
-            self['deleteCacheOnExit'] = TempDirectory(idir)
+            # Alan Malta in 29 Mar 2022: this seems to prematurely remove the cache
+            # directory. For details, see: https://github.com/dmwm/WMCore/pull/10915
+            # self['deleteCacheOnExit'] = TempDirectory(idir)
```

Comment:
it is unclear to me how this change will affect the rest of the running services/app. If you do not delete temp directory who will take care of it? And, how? I rather prefer to see full description how this should be done. And, so far this PR does not address this. You may well been fixed the current issue but not handling temp area may cause another one, when eventually you'll fill out the disk with non empty temp directories.

Reply:
The `/tmp` directory has an intrinsic temporary nature, reason why we actually do not use /tmp for our services deployed in CMSWEB. In other words, at least in Unix systems, we cannot guarantee that a temporary file/directory will remain available for us.

For further details, please have a look at the first description of this PR. It could be that I updated it while you were looking at it.

Reply:
I'm not sure it answer my question. The cache is not related to `/tmp` in any way, but you commented out `dleteCacheonExit` key. Based on its name I would conclude that after some operations the cache is not cleaned. As with any disk space it will keep growing until someone will do a clean-up. Therefore my question is not about `TempDirectory` or `/tmp` but rather how the cache is cleaned and where. A simple statement that we never experience an issue it does not mean that eventually disk will fill out and you will end-up with the filled disk issue. What I asked is to clarify how cache clean-up is handled.